src/app/data/services/users.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable, inject } from '@angular/core';
+import { User } from '../intefaces/user.interface';
+import { UsersApiService } from './users-api-service.service';
+import { BehaviorSubject } from 'rxjs';
+import { LocalStorageService } from './local-storage.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UsersService {
+  userApiSevice = inject(UsersApiService);
+  private readonly localStorageService = inject(LocalStorageService);
+  private usersSubject$ = new BehaviorSubject<User[]>(
+    this.localStorageService.getUsersLocalStorage()
+  );
+  public readonly users$ = this.usersSubject$.asObservable();
+
```

Comment:
`constructor() {
   this.usersSubject$.subscribe(result => this.localStorageService.saveUsersToLocalStorage(result))
}`